lib/OpenQA/CacheService/Model/Cache.pm

```diff
@@ -47,139 +48,106 @@ sub from_worker {
-    $ua->max_response_size(0);
+    my $ua  = Mojo::UserAgent->new(max_redirects => 2, max_response_size => 0);
     my $url = sprintf '%s/tests/%d/asset/%s/%s', $self->host, $id, $type, basename($asset);
-    log_info "Downloading " . basename($asset) . " from $url";
-    my $tx = $ua->build_tx(GET => $url);
-    my $headers;
-
-    $ua->on(
-        start => sub {
-            my ($ua, $tx) = @_;
+    $log->info('Downloading ' . basename($asset) . " from $url");
 
-            my $progress     = 0;
-            my $last_updated = time;
-            if (-e $asset) {    # Assets might be deleted by a sysadmin
-                $tx->req->headers->header('If-None-Match' => qq{$etag}) if $etag;
-            }
-            $tx->res->on(
-                progress => sub {
-                    my $msg = shift;
-                    $msg->finish if $msg->code == 304;
-                    return unless my $len = $msg->headers->content_length;
-
-                    my $size = $msg->content->progress;
-                    $headers = $msg->headers if !$headers;
-                    my $current = int($size / ($len / 100));
-
-                    # Don't spam the webui, update only every 5 seconds
-                    if (time - $last_updated > 5) {
-                        $last_updated = time;
-                        if ($progress < $current) {
-                            $progress = $current;
-                            log_debug "CACHE: Downloading $asset: " . ($size == $len ? 100 : $progress) . "%";
-                        }
-                    }
-                });
-        });
+    # Assets might be deleted by a sysadmin
+    my $tx = $ua->build_tx(GET => $url);
+    $tx->req->headers->header('If-None-Match' => qq{$etag}) if $etag && -e $asset;
 
     $tx = $ua->start($tx);
-    my $code = ($tx->res->code) ? $tx->res->code : 521;    # Used by cloudflare to indicate web server is down.
-    my $size;
+    my $res = $tx->res;
+    my $ret;
+
+    my $code = $res->code // 521;    # Used by cloudflare to indicate web server is down.
     if ($code eq 304) {
-        if ($self->_update_asset_last_use($asset)) {
-            log_debug "CACHE: Content has not changed, not downloading the $asset but updating last use";
-        }
-        else {
-            log_debug 'CACHE: Abnormal situation, code 304. Retrying download';
-            $asset = 520;
+        $log->info("Content of $asset has not changed, updating last use");
+        unless ($self->_update_asset_last_use($asset)) {
+            $log->warn('Abnormal situation, status 304');
+            $ret = 520;
         }
     }
-    elsif ($tx->res->is_server_error) {
-        log_debug "CACHE: Could not download the asset, triggering a retry for $code.";
-        log_debug "CACHE: Abnormal situation, server error. Retrying download";
-        $asset = $code;
+
+    elsif ($res->is_server_error) {
+        $log->warn("Abnormal situation, server error $code");
```

Comment:
Also here "Abnormal situation" is too generic. I had took up what `is_server_error` means but it is merely a check for 5xx error codes. So maybe just "Error downloading asset, server responded error $code". And the server is the web UI here, right? So maybe even "Error downloading asset, web UI responded error $code"?

Reply:
Done.